tc/tc_util.c

```diff
@@ -191,6 +191,61 @@ static const struct rate_suffix {
 	{ NULL }
 };
 
+int parse_percent_rate(char *rate, const char *str, char *dev)
+{
+	long max_rate_bits;
+	int ret;
+	double perc, rate_bits;
+
+	if (dev == NULL) {
```

Comment:
Since we declare the space for the device name on the stack in `tc_qdisc_modify()`, `dev` will never be `NULL`. Instead, the test should be `if (!dev[0]) {`.